etc/afpd/afpstats.c

```diff
@@ -22,31 +22,74 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <dbus/dbus-glib.h>
 #include <glib.h>
+#include <gio/gio.h>
 
 #include <atalk/compat.h>
 #include <atalk/errchk.h>
 #include <atalk/logger.h>
 #include <atalk/server_child.h>
 
 #include "afpstats_obj.h"
-#include "afpstats_service_glue.h"
+//#include "afpstats_service_glue.h"
 
 /*
  * Beware: this struct is accessed and modified from the main thread
  * and from this thread, thus be careful to lock and unlock the mutex.
  */
 static server_child_t *childs;
+static GDBusNodeInfo *introspection_data = NULL;
+
+static void handle_method_call (GDBusConnection *connection,
+                    const gchar           *sender,
+                    const gchar           *object_path,
+                    const gchar           *interface_name,
+                    const gchar           *method_name,
+                    GVariant              *parameters,
+                    GDBusMethodInvocation *invocation,
+                    gpointer               user_data)
+{
+}
```

Comment:
You need an implementation here, which should look a bit like the one in https://gitlab.gnome.org/GNOME/glib/-/blob/main/gio/tests/gdbus-example-server.c?ref_type=heads (but it can be a lot simpler, because it seems like your interface only has one method).

Probably something like this (untested!):

```suggestion
static void handle_method_call (GDBusConnection *connection,
                    const gchar           *sender,
                    const gchar           *object_path,
                    const gchar           *interface_name,
                    const gchar           *method_name,
                    GVariant              *parameters,
                    GDBusMethodInvocation *invocation,
                    gpointer               user_data)
{
  AFPStatsObj *obj = AFPSTATS_OBJECT (user_data);

  if (g_strcmp0 (method_name, "GetUsers") == 0)
    {
      GError *local_error = NULL;
      gchar **users = NULL;
      
      if (afpstats_obj_get_users (obj, &users, &local_error))
        g_dbus_method_invocation_return_value (invocation, g_variant_new ("(^as)", users)));
      else
        g_dbus_method_invocation_return_gerror (invocation, local_error);
        
      g_strfreev (users);
      g_clear_error (&error);
    }
}
```